apps/desktop-wallet/src/api/apiDataHooks/wallet/useFetchWalletAddressesActivityTimestampsSorted.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { orderBy } from 'lodash'
+import { useMemo } from 'react'
+
+import { SkipProp } from '@/api/apiDataHooks/apiDataHooksTypes'
+import { useFetchWalletActivityTimestamps } from '@/api/apiDataHooks/wallet/useFetchWalletLastTransactions'
+import { useAppSelector } from '@/hooks/redux'
+import { selectDefaultAddress } from '@/storage/addresses/addressesSelectors'
+
+const useFetchWalletAddressesActivityTimestampsSorted = (props?: SkipProp) => {
```

Comment:
I'm confused. We have 2 very close hooks, this one and `useFetchWalletAddressesSortedByActivity`. 👀
Just comparing the names I can't tell the difference. I see that the latter rely on this one, and the later follow our naming convention but not this one. 
Looking at the code it seems that all `useFetchWalletAddressesSortedByActivity` do is removing the timestamps and returning just an array of AddressHashes.

There's probably something I'm missing, but as of now I'm not sure why 2 different hooks are necessary 🤔

Reply:
I understand your confusion. Let me explain:

> TL;DR:
> - `useFetchWalletAddressesSortedByActivity` is a 1-to-1 replacement for `useAppSelector(selectAllAddressHashes)`
> - `useFetchWalletAddressesActivityTimestampsSorted` is a more general hook that includes the timestamps

In many components we need to get the list of addresses of the wallet. Until now, we've been relying on Redux for that using:

```ts
const allAddressHashes = useAppSelector(selectAllAddressHashes)
```

The above was taking care of sorting the addresses due to the `lastUsed` property that the `Address` Redux slice entity had. The sorting was defined in the adapter. Now that all API-related data have been removed from Redux, we need to use Tanstack for sorting. Hence the creation of `useFetchWalletAddressesSortedByActivity` which returns **exactly** the same result as `useAppSelector(selectAllAddressHashes)`.

Another feature of the above hook is that until we have the last used timestamps data for ALL addresses, it will return the unsorted list, otherwise it would return a different result the moment a new timestamp is fetched, which would create a lot of jumping around in the UI:

```ts
data: !isLoading && !props?.skip ? sortedAddressHashes : allAddressHashes
```

The above `useFetchWalletAddressesSortedByActivity` hook is based on `useFetchWalletAddressesActivityTimestampsSorted`. The latter is a more basic hook that does not have the above "loading" feature that I mentioned, and it also returns the timestamp together with each address which is useful in the case of the bulk address delete where we preselect the addresses that have never been used.

Do you have another idea to achieve the desired results? Maybe with different names?

Reply:
Much better after the renames we talked about!